core_lib/src/tool/selecttool.cpp

```diff
@@ -39,57 +39,58 @@ void SelectTool::loadSettings()
 
 QCursor SelectTool::cursor()
 {
-    MoveMode mode = mScribbleArea->getMoveModeForSelectionAnchor();
-    return mScribbleArea->currentTool()->selectMoveCursor(mode, type());
+    MoveMode mode = mEditor->select()->getMoveModeForSelectionAnchor(getCurrentPoint());
+    BaseTool* tool = mEditor->tools()->currentTool();
+    return tool->selectMoveCursor(mode, type());
```

Comment:
Wouldn't `tool` end up just being `this`?

Reply:
Indeed, that's a brainfart :P